res/controllers/Hercules_DJControl_Jogvision-scripts.js

```diff
@@ -0,0 +1,262 @@
+// * Author: DJ Phatso, contributions by Kerrick Staley and David TV
+// * Version 1.3 (November 2019)
+// * Version 1.2 (November 2019)
+// * Version 1.1 (March 2019)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12580
+// * Wiki: https://www.mixxx.org/wiki/doku.php/hercules_dj_control_jogvision
+//
+// Changes to v1.3
+// - Enabled the creation of beatloop (beatloop_activate) by using SHIFT+LoopON
+// - Changed "LOOP SIZE" to adjust the beatjump_size (you still can change size
+//   with surrounding buttons)
+// - Added SHIFT+"LOOP SIZE" to move loop left or right by N beats (beatjump)
+//
+// Changes to v1.2
+// - Enabled Jogwheel Outer LED rotation
+// - Enabled Beat LEDs
+// 
+// Changes to v1.1
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+// 
+// v1.0 : Original release
+//
+// ****************************************************************************
+
+var on = 0x7F;
+var off = 0x00;
+var alpha = 1.0 / 8;
+var beta = alpha / 16;
+var ledRotationSpeed = 4;
+var ledRotationTimer;
+
+var DJCJV = {};
+
+DJCJV.scratchButtonState = true;
+DJCJV.Channel=[];
+DJCJV.Channel['[Channel1]'] = { "central":0x90, "deck":0xB0, "beatPosition":1, "rotation":0x00, "n":1 };
+DJCJV.Channel['[Channel2]'] = { "central":0x91, "deck":0xB1, "beatPosition":1, "rotation":0x00, "n":2 };
+
+// Initialization
+DJCJV.init = function() {
+
+	//Set all LED states to off
+	midi.sendShortMsg(0xB0, 0x7F, off);
+	midi.sendShortMsg(0xB1, 0x7F, off);
+
+	// Set Vinyl button LED On.
+	midi.sendShortMsg(0x90, 0x45, on);
+	DJCJV.scratchButtonState = true;
+	midi.sendShortMsg(0x90, 0x46, on);
+
+	// Set Headphone CUE/MIX LED state
+	if (engine.getValue("[Master]", "headMix") > 0.5) {
+		midi.sendShortMsg(0x90, 0x4C, on); // headset "Mix" button LED
+		midi.sendShortMsg(0x90, 0x4D, off);
+	}
+	else {
+		midi.sendShortMsg(0x90, 0x4C, off);
+		midi.sendShortMsg(0x90, 0x4D, on); // headset "Cue" button LED
+	}
+
+	//Enable Soft takeover
+	engine.softTakeover("[Master]", "crossfader", true);
+	engine.softTakeover("[QuickEffectRack1_[Channel1]]", "super1", true);
+	engine.softTakeover("[QuickEffectRack1_[Channel2]]", "super1", true);
+
+	//Set effects Levels - Dry/Wet - Filters
+	engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit1_Effect2]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit1_Effect3]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit1]", "mix", 0.6);
+	engine.setParameter("[EffectRack1_EffectUnit2]", "mix", 0.6);
+	engine.setParameter("[QuickEffectRack1_[Channel1]]", "super1", 0.5);
+	engine.setParameter("[QuickEffectRack1_[Channel2]]", "super1", 0.5);
```

Comment:
Why do you mass-set these values? I think that this would cause issues if Mixxx gets MIDI hotplug support in the future. If a user accidently disconnects the controller and then reconnects it, this could cause abrupt changes in the audio output.

Reply:
This was done in the original script to "work around" the limited controls available on the Jogvision (which were built out of the bundle with Serato).

Since there is only one encoder that sends 3 midi signals to control the FX amount (meaning it would affect the level of all 3 effects simultaneously), I figured I could instead configure the fx rack as auto-FX for all  3 buttons and preset the levels to an appropriate amount for each of them. 

The same goes for the motion sensor "Air FX", which goes back to 0 by default. In order to use if with the Filter, it needed to be set to the middle position, and it was decided that only the lower range would be used.

There is surely a more practical and elegant ways to do all this for sure, but my coding knowledge are still limited.  Any suggestions are much welcomed.